scRNA-seq/02-normalizing_scRNA-seq.Rmd

```diff
@@ -67,185 +66,298 @@ sc_metadata <- readr::read_tsv(metadata_file)
+
+```{r model_variance}
+gbm_variance <- modelGeneVar(gbm_sce)
+```
+
+```{r plot_variance}
+ggplot(as.data.frame(gbm_variance), aes(x= mean, y = total)) +
+  geom_point(alpha = 0.05) +
+  stat_function(fun = metadata(gbm_variance)$trend, color = "blue") + 
+  xlab("Mean log-expression") + 
+  ylab("Variance")
+
+```
+
+```{r get_highvar}
+highvar_genes <- scran::getTopHVGs(gbm_variance, n = 500)
+highvar_pca <- scater::calculatePCA(gbm_sce, subset_row = highvar_genes)
+```
+
+```{r highvar_df}
+highvar_pca_scores <- data.frame(highvar_pca,
+                                 geo_accession = rownames(highvar_pca),
+                                 cell_type = gbm_sce$cell_type)
+```
+
+```{r plot_highvar}
+ggplot(highvar_pca_scores, aes(x = PC1, y = PC2, color = cell_type)) +
+  geom_point() + 
+  ggtitle("High variance gene PCA scores") + 
+  colorblindr::scale_color_OkabeIto()
+```
+
+Now what we didn't tell you before, but will tell you now, is that the default for `calculatePCA()` is to use the top 500 highest variance genes. 
+This is what we were overriding when we used `ntop = nrow(gbm_sce)`. 
+So if we want to shortcut what we did to get the above plot (or one very close to it), we could have done it with `scater::calculatePCA(gbm_sce)` using no additional arguments.
+
+*Why didn't we do that first?*
+Well, it wouldn't have shown you the effects of normalization or making the choice to include only high variance genes. 😉
+
+
+## Identify marker genes
+
+The highest variance genes may not always be the most informative on their own, especially if we are trying to differentiate among cell types.
+To find genes that are distinct to one cell type or another, we can use the `scran::findMarkers()` function to test for genes that are significantly different among cells of each type.
+This will return a list of tables, one for each cell type, with statistics for each gene showing how well it differentiates that cell type against all other types.
+
+```{r find_markers}
+markers <- scran::findMarkers(gbm_sce, gbm_sce$cell_type)
+```
+
+Next we can look at one of those tables, in this case the one for neurons. 
+We have done some transformation here to pull the gene name into a column of its own and filtered to the genes that seem to differentiate best using the `Top` column.
+
+```{r marker_table}
+markers$Neuron %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("gene") %>%
+  dplyr::filter(Top <= 100) # most significant 100 for each pairwise comparison
+
+```
+
+Because we tend to like tidy data, here we use a `tidyverse` function from the [`purrr` package](https://purrr.tidyverse.org) to apply the same operations as above to every element of the `markers` list. 
+`purrr::map_df` is particularly nice, because after applying the operations in the second argument, it combines the results into a single data frame, adding a column with the name specified by `.id` to differentiate the rows that came from each element of the list.
```

Comment:
```suggestion
`purrr::map_df()` is particularly nice, because after applying the operations in the second argument, it combines the results into a single data frame, adding a column with the name specified by `.id` to differentiate the rows that came from each element of the list.
```